framework/src/media/audio/audio_manager.c

```diff
@@ -191,15 +190,12 @@ static audio_manager_result_t find_audio_card(audio_card_type_t card_type)
 			snprintf(card[card_id].card_path, strlen("/dev/audio/") + strlen(dir_entry->d_name) + 1, "%s%s", "/dev/audio/", dir_entry->d_name);
 			card[card_id].status = AUDIO_CARD_IDLE;
 			card[card_id].device_id = device_id;
-			found_cards++;
 			medvdbg("Found an audio card, total card : %d id : %d, card_path : %s\n", found_cards, card_id, card[card_id].card_path);
-			get_hardware_params(&card[card_id], card_type);
 		}
-
 	}
 
 	/* If nothing founded */
-	if (found_cards == 0) {
+	if (card_id == INVALID_ID) {
```

Comment:
If you want card_id then you should compare device_id too, because if all cards contains wrong device_id value, this will failed, that's why I used found_cards.

Reply:
Cool. I'll revert this modification.